src/internal/unix/no-shell.js

```diff
@@ -18,7 +20,9 @@ const unsupportedError = "Quoting is not supported when no shell is used";
  * @returns {string} The escaped argument.
  */
 function escapeArg(arg) {
-  return arg.replace(/[\0\u0008\u001B\u009B]/gu, "").replace(/\r(?!\n)/gu, "");
+  const controlCharacters = new RegExp("[\0\u0008\u001B\u009B]", "g");
+  const carriageReturns = new RegExp("(?:(\r\n)|\r)", "g");
+  return arg.replace(controlCharacters, "").replace(carriageReturns, "$1");
```

Comment:
Given the [substantial overhead of using `new RegExp()`](https://github.com/ericcornelissen/shescape/issues/2122#issuecomment-3198102796) we'd want to avoid instantiating the expression with every call (applies to all uses of `new RegExp` in this Pull Request).